code/game/gamemodes/traitor/double_agents.dm

```diff
@@ -18,29 +22,179 @@
+		return
+	var/already_traitored = traitored
+	to_chat(owner.current, "<B><font size=3 color=red> Target eliminated: [victim.name]</font></B>")
+	for(var/objective_ in victim.objectives)
+		if(istype(objective_, /datum/objective/assassinate/internal))
+			var/datum/objective/assassinate/internal/objective = objective_
+			if(objective.target==owner)
+				traitored = TRUE
+			else if(targets_stolen.Find(objective.target) == 0)
+				var/datum/objective/assassinate/internal/new_objective = new
+				new_objective.owner = owner
+				new_objective.target = objective.target
+				new_objective.update_explanation_text()
+				owner.objectives += new_objective
+				targets_stolen += objective.target
+				var/status_text = objective.check_completion() ? "neutralised" : "active"
+				to_chat(owner.current, "<B><font size=3 color=red> New target added to database: [objective.target.name] ([status_text]) </font></B>")
+		else if(istype(objective_, /datum/objective/destroy/internal))
+			var/datum/objective/destroy/internal/objective = objective_
+			var/datum/objective/destroy/internal/new_objective = new
+			if(objective.target==owner)
+				traitored = TRUE
+			else if(targets_stolen.Find(objective.target) == 0)
+				new_objective.owner = owner
+				new_objective.target = objective.target
+				new_objective.update_explanation_text()
+				owner.objectives += new_objective
+				targets_stolen += objective.target
+				var/status_text = objective.check_completion() ? "neutralised" : "active"
+				to_chat(owner.current, "<B><font size=3 color=red> New target added to database: [objective.target.name] ([status_text]) </font></B>")
+	if(traitored&&!already_traitored)
+		for(var/objective_ in owner.objectives)
+			if(!is_internal_objective(objective_))
+				continue
+			var/datum/objective/assassinate/internal/objective = objective_
+			if(!objective.check_completion())
+				traitored = FALSE
+				return
+		to_chat(owner.current,"<B><font size=3 color=red> All the other agents are dead, and you're the last loose end. Stage a Syndicate terrorist attack to cover up for today's events. You no longer have any limits on collateral damage.</font></B>")
+		
+			
+	
+/datum/internal_agent_state/proc/steal_targets_timer_func()
+	if(owner&&owner.current&&owner.current.stat!=DEAD)
+		for(var/objective_ in owner.objectives)
+			if(!is_internal_objective(objective_))
+				continue
+			var/datum/objective/assassinate/internal/objective = objective_
+			if(!objective.target)
+				continue
+			if(objective.check_completion())
+				if(objective.stolen)
+					continue
+				else
+					steal_targets(objective.target)
+					objective.stolen = TRUE
+			else
+				if(objective.stolen)
+					var/fail_msg = "<B><font size=3 color=red>Your sensors tell you that [objective.target.current.real_name], one of the targets you were meant to have killed, pulled one over on you, and is still alive - do the job properly this time! </font></B>"

```

Comment:
instead of `<B><font size=3 color=red>`, use `<span class='userdanger'>`
do this for all places this appears.